frontend/alert/src/components/AutoComplete.js

```diff
@@ -0,0 +1,237 @@
+    z-index: 1;
+`;
+
+const AutoCompleteInputBackground = styled(AutoCompleteInput)`
+    background: white;
+    z-index: 0;
+    color: grey;
+`;
+
+const AutoCompleteContainer = styled.div`
+    position: relative;
+    display: block;
+    margin-bottom: 1rem;
+    left: -50%;
+    height: ${props => props.inputHeight};
+`;
+
+const Suggestion = ({ courseCode, title, instructor }) => (
+    <SuggestionBox>
+        <SuggestionLeftCol>
+            <SuggestionTitle>{courseCode}</SuggestionTitle>
+            <SuggestionSubtitle>{title}</SuggestionSubtitle>
+            <SuggestionSubtitle>{instructor}</SuggestionSubtitle>
+        </SuggestionLeftCol>
+        <IconContainer>
+            <FontAwesomeIcon icon={faHistory} color="#c4c4c4" />
+        </IconContainer>
+    </SuggestionBox>
+);
+
+Suggestion.propTypes = {
+    courseCode: PropTypes.string,
+    title: PropTypes.string,
+    instructor: PropTypes.string,
+};
+
+/**
+ * Given a current input value and a list of suggestions, generates the backdrop text for the
+ * autocomplete input
+ * @param value
+ * @param suggestions
+ * @return {string|*}
+ */
+const generateBackdrop = (value, suggestions) => {
+    if (!value || !suggestions || !suggestions[0]) {
+        return "";
+    }
+    let suggestion = suggestions[0].section_id;
+    const valueIsLower = value.charAt(0)
+        .toLowerCase() === value.charAt(0);
+    if (valueIsLower) {
+        suggestion = suggestion.toLowerCase();
+    }
+    // prevent issues where a lag in suggestions updating causes weird overlap
+    suggestion = value + suggestion.substring(value.length);
+    return suggestion;
+};
+
+const AutoComplete = () => {
+    const [inputRef, setInputRef] = useState(null);
+    const [value, setValue] = useState("");
+    const [suggestions, setSuggestions] = useState([]);
+    const [active, setActive] = useState(false);
+    const [backdrop, setBackdrop] = useState("");
+
+    const show = active && suggestions.length > 0;
+
+    useEffect(() => {
+        setBackdrop(generateBackdrop(
+            inputRef && value,
+            show && suggestions
+        ));
+    }, [inputRef, show, suggestions, value]);
+
+    useEffect(() => {
+        if (!value) {
+            setSuggestions([]);
+        } else {
+            suggestionsDebounced(value)
+                .then(({ searchResult, searchTerm }) => {
+                    // make sure the search term is not stale
+                    if (searchTerm === value) {
+                        setSuggestions(searchResult);
+                    }
+                });
+        }
+    }, [value]);
+
+
+    return (
+        <AutoCompleteContainer
+            inputHeight={inputRef && `${inputRef.getBoundingClientRect().height}px`}
+            ref={useOnClickOutside(() => setActive(false), !show)}
+        >
+            <AutoCompleteInput
+                autocomplete="off"
+                placeholder="Course"
+                ref={setInputRef}
```

Comment:
Might be able to simplify this with the `useRef` hook to simplify this a bit: https://reactjs.org/docs/hooks-reference.html#useref

Reply:
I had it as that initially, but it didn't work because it didn't force rerenders.